File: apps/mobile/lib/domain/entities/tag.dart
class Tag {
  final int id;
  final int? groupId;
  final int createdBy;
  final String name;
  final DateTime createdAt;
  final DateTime updatedAt;

  Tag({
    required this.id,
    this.groupId,
    required this.createdBy,
    required this.name,
    required this.createdAt,
    required this.updatedAt,
  });

  Tag copyWith({
    int? id,
    int? groupId,
    int? createdBy,
    String? name,
    DateTime? createdAt,
    DateTime? updatedAt,
  }) {
    return Tag(
      id: id ?? this.id,
      groupId: groupId ?? this.groupId,
      createdBy: createdBy ?? this.createdBy,
      name: name ?? this.name,
      createdAt: createdAt ?? this.createdAt,
      updatedAt: updatedAt ?? this.updatedAt,
    );
  }
}

File: apps/api/src/domain/models/tag.rs
use chrono::NaiveDateTime;
use serde::{Deserialize, Serialize};
use sqlx::FromRow;

#[derive(Debug, Clone, Serialize, Deserialize, FromRow)]
pub struct Tag {
    pub id: i64,
    pub group_id: Option<i64>,
    pub created_by: i64,
    pub name: String,
    #[sqlx(default)]
    pub created_at: NaiveDateTime,
    #[sqlx(default)]
    pub updated_at: NaiveDateTime,
}

